src/tools/rbd_mirror/image_sync/SnapshotCopyRequest.h

```diff
@@ -104,6 +104,7 @@ class SnapshotCopyRequest : public BaseRequest {
   librados::snap_t m_prev_snap_id = CEPH_NOSNAP;
 
   std::string m_snap_name;
+  cls::rbd::SnapshotNamespace m_snap_namespace;
```

Comment:
Do you mean you want me to declare only UserSnapshotNamespace here? Then when I retrieve a snapshot_namespace from SnapInfo I should check that what arrived is actually UserSnapshotNamespace. How should the code react if the namespace that arrived is not UserSnapshotNamespace? Is an assert(snap_info.snapshot_namespace == UserSnapshotNamespace) enough or you want a more intelligent and verbose handling?